cetmix_tower_server_notify_backend/models/cx_tower_plan_log.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class CxTowerPlanLog(models.Model):
+    _inherit = "cx.tower.plan.log"
+
+    def _plan_finished(self):
+        res = super()._plan_finished()
+
+        context_timestamp = fields.Datetime.context_timestamp(
+            self, fields.Datetime.now()
+        )
+        for log in self:
+            # don't notify if a plan that was runned from another plan has been executed
+            if log.parent_flight_plan_log_id:
+                continue
+
+            if log.plan_status == 0:
+                log.create_uid.notify_success(
```

Comment:
I propose we refactor our log model to include a new field:

```python

user_id = fields.Many2one('res.users', string="Author", index=True)
```
Why?
Right now, when we run flight plans or other operations with `sudo(),` the Odoo records the superuser as the `create_uid`. That means any notifications or access checks tied to the log entry will go to the superuser, not the actual user who triggered the action. By adding an explicit user_id field for the “real” author, we can:

Accurately track who initiated each action, even if it runs with elevated privileges.

Send notifications to the correct user, rather than always to the superuser.

Audit user activity more reliably in multi-user environments.

@ivs-cetmix what do you think?

Reply:
@GabbasovDinar I think this is not an issue, because `sudo()` doesn't replace the user with a superuser any more. It was introduced in Odoo 12.0 or 13.0 afaik.